W/Wayland/build_tarballs.jl

```diff
@@ -3,29 +3,51 @@
 using BinaryBuilder
 
 name = "Wayland"
-version = v"1.21.0"
+version = v"1.23.0"
 
 # Collection of sources required to build Wayland
 sources = [
     ArchiveSource("https://gitlab.freedesktop.org/wayland/wayland/-/releases/$(version)/downloads/wayland-$(version).tar.xz",
-                  "6dc64d7fc16837a693a51cfdb2e568db538bfdc9f457d4656285bb9594ef11ac"),
+                  "05b3e1574d3e67626b5974f862f36b5b427c7ceeb965cb36a4e6c2d342e45ab2"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/wayland-*/
 
-ln -s `which wayland-scanner` $bindir
-cp $prefix/libdata/pkgconfig/* $prefix/lib/pkgconfig || true
+# Build a native version of wayland-scanner
 
-mkdir build-wayland
+cat - >host_pkgconfig.meson <<EOF
+[binaries]
+pkgconfig = ['/usr/bin/pkg-config', '--define-variable=prefix=${host_prefix}']
+EOF
```

Comment:
Is this something that should be added to the default [host meson toolchain file](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/master/src/BuildToolchains.jl)?

Reply:
Maybe? The problem I'm working around here is that `Expat_jll`'s `expat.pc` file hard-codes a value for `prefix` which points to a different path than the path use for `HostBuildDependency`s.

The other thing I found was [relocatable](https://manpages.debian.org/stretch/pkg-config/pkg-config.1.en.html#PKG-CONFIG_DERIVED_VARIABLES) pkg-config files where `prefix` is instead set to something like
```
prefix=${pcfiledir}/../..
```
which if all dependencies were patched to do this would presumably make the override unnecessary.

Reply:
In the most recent version, I've switched this to use `'--define-variable=prefix=\${pcfiledir}/../..'` which also works and opts in to the relocatable version rather than choosing either the host or target path. I think with that change, this kind of change could be a candidate addition to various toolchain files.